Dockerfile

```diff
@@ -0,0 +1,9 @@
+# ----- Base -----
+FROM node:16
+WORKDIR /usr/app
+
+# ----- Install & Build -----
+COPY ["package.json", "package-lock.json", "babel.config.json", "./"]
+RUN npm install
+COPY . .
+RUN npm run build
```

Comment:
It seems that the only real benefit you get from Docker in your current setup with this line and volume config in docker-compose is building your dependencies in order to build source code (TS->JS).
And that's only possible because you're sharing your root via volume with that a container.
Your app is still running outside Docker.

Therefore, a couple of suggestions:
1. Start by editing this Dockerfile to run your app as well in a container in Dev 
2. Create a similar Dockerfile for prod (consider changing some lines that would make sense for prod), you'll end up with 2 Dockerfiles in your root, so naming is important (docker-compose context hint here :) )
3. Update docker-compose to use different Dockerfiles depending on env variable

Then refactor a bit to try another way of doing things:
1. Get rid of dev Dockerfile and rename your prod Dockerfile back to be a single Dockerfile in root
2. Improve your Dockerfile further by making npm run script dynamic (either dev or prod) allowing to choose the env externally (e.g via build-args).
3. Update your docker-compose to pass those external args to your image

Optionally, look into how you can leverage Multi-Stage Builds in your Dockerfile